File: Q7.java
public class Q7{

    public int reverse(int x){
	int reversed = 0;
        
	while (x != 0){
	    // First pop using a modulo and int division to trim
	    int pop = x % 10;
	    x /= 10;

	    // Checks for out of bounds
	    if (reversed >= Integer.MAX_VALUE/10 || reversed <= Integer.MIN_VALUE/10) return 0;

	    // Then the pop integer needs to be added to reversed
	    reversed = reversed * 10 + pop;
	}

	return reversed;
    }

}

/*
Runtime: 1 ms, faster than 100.00% of Java online submissions for Reverse Integer.
Memory Usage: 33.5 MB, less than 11.66% of Java online submissions for Reverse Integer.
*/

File: Q7/Q7.py
class Solution:
    def reverse(self, x):
        str_x = str(x)
        new_str = ""

        # Flip the string. 
        for i in range(0, len(str_x)):
            new_str = new_str + str_x[len(str_x)-1-i]

        new_str = new_str.replace("-", "")
        
        # If negative add sign
        if x < 0:
            new_str = "-" + new_str

        if int(new_str) > 2**31 - 1 or int(new_str) < -2**31:
            return 0
        else:
            return int(new_str)

"""
Runtime: 36 ms, faster than 84.51% of Python3 online submissions for Reverse Integer.
Memory Usage: 13.9 MB, less than 5.26% of Python3 online submissions for Reverse Integer.
"""

